src/storage/compression/bitpacking.cpp

```diff
@@ -257,7 +257,7 @@ struct BitpackingState {
 			    BitpackingPrimitives::MinimumBitWidth<T, false>(static_cast<T>(min_max_delta_diff));
 			auto regular_required_bitwidth = BitpackingPrimitives::MinimumBitWidth(min_max_diff);
 
-			if (delta_required_bitwidth < regular_required_bitwidth && mode != BitpackingMode::FOR) {
+			if ((!can_do_for || delta_required_bitwidth < regular_required_bitwidth) && mode != BitpackingMode::FOR) {
```

Comment:
Just writing this out for myself, to verify that I understand the change.

`can_do_for` gets determined by `TrySubtractOperator`
So that means if the difference between `minimum` and `maximum` is too large, it'll be false.

As you said, `regular_required_bitwidth` will be 0 when `can_do_for` is false (because it gets initialized to that), so `delta_required_bitwidth` will never be smaller than `regular_required_bitwidth` if `can_do_for` is false.

So we're essentially just checking whether `min_max_diff` is initialized to a real value, by checking `can_do_for`
I had a hard time reasoning about the condition in the way that it's written though, can we flip this to `(can_do_for && delta_required_bitwidth < regular_required_bitwidth)` ?

Reply:
@xuke-hat 
I had a hard time reasoning about the condition in the way that it's written though, can we flip this to `(can_do_for && delta_required_bitwidth < regular_required_bitwidth)` ?